This fixes the wlshm path which was not updating the buffer correctly.
---
 hw/xfree86/xwayland/xwayland.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/hw/xfree86/xwayland/xwayland.c b/hw/xfree86/xwayland/xwayland.c
index d97f4ee..f59bfe4 100644
--- a/hw/xfree86/xwayland/xwayland.c
+++ b/hw/xfree86/xwayland/xwayland.c
@@ -344,6 +344,9 @@ void xwl_screen_post_damage(struct xwl_screen *xwl_screen)
 			      box->x2 - box->x1 + 1,
 			      box->y2 - box->y1 + 1);
 	}
+	wl_surface_attach(xwl_window->surface,
+			  xwl_window->buffer,
+			  0, 0);
 	wl_surface_commit(xwl_window->surface);
 	DamageEmpty(xwl_window->damage);
     }
-- 
1.7.10.4 
